src/AppInstallerCommonCore/GroupPolicy.cpp

```diff
@@ -0,0 +1,186 @@
+            {
+                return PolicyState::NotConfigured;
+            }
+
+            auto togglePolicy = TogglePolicyInternal::GetPolicy(policy);
+
+            // Policies are not configured if there is no registry value.
+            auto setting = RegistryValueIsTrue(key, togglePolicy.RegValueName);
+            if (!setting.has_value())
+            {
+                return PolicyState::NotConfigured;
+            }
+
+            // Return flag as-is or invert depending on the policy
+            return *setting ? PolicyState::Enabled : PolicyState::Disabled;
+        }
+
+        template <ValuePolicy P>
+        void Validate(
+            const Registry::Key& policiesKey,
+            GroupPolicy::ValuePoliciesMap& policies)
+        {
+            auto value = details::ValuePolicyMapping<P>::ReadAndValidate(policiesKey);
+            if (value.has_value())
+            {
+                policies.Add<P>(std::move(*value));
+            }
+        }
+
+        template <size_t... P>
+        void ValidateAllValuePolicies(
+            const Registry::Key& policiesKey,
+            GroupPolicy::ValuePoliciesMap& policies,
+            std::index_sequence<P...>)
+        {
+            // Use folding to call each policy validate function.
+            (FoldHelper{}, ..., Validate<static_cast<ValuePolicy>(P)>(policiesKey, policies));
+        }
+    }
+
+    namespace details
+    {
+        std::optional<uint32_t> ValuePolicyMapping<ValuePolicy::SourceAutoUpdateIntervalInMinutes>::ReadAndValidate(const Registry::Key& policiesKey)
+        {
+            using Mapping = ValuePolicyMapping<ValuePolicy::SourceAutoUpdateIntervalInMinutes>;
+            return GetRegistryValue<Mapping::ValueType>(policiesKey , Mapping::ValueName);
+        }
+
+        std::optional<std::vector<std::string>> ValuePolicyMapping<ValuePolicy::IncludeSources>::ReadAndValidate(const Registry::Key&)
+        {
+            // TODO
+            return std::nullopt;
+        }
+    }
+
+    GroupPolicy::GroupPolicy(const Registry::Key& key)
+    {
+        ValidateAllValuePolicies(key, m_values, std::make_index_sequence<static_cast<size_t>(ValuePolicy::Max)>());
+
+        using Toggle_t = std::underlying_type_t<TogglePolicy>;
+        for (Toggle_t i = static_cast<Toggle_t>(TogglePolicy::None); i < static_cast<Toggle_t>(TogglePolicy::Max); ++i)
+        {
+            auto policy = static_cast<TogglePolicy>(i);
+            m_toggles[policy] = GetStateInternal(key, policy);
+        }
+    }
+
+    PolicyState GroupPolicy::GetState(TogglePolicy policy) const
+    {
+        auto itr = m_toggles.find(policy);
+        if (itr == m_toggles.end())
+        {
+            return PolicyState::NotConfigured;
+        }
+
+        return itr->second;
+    }
+
+    bool GroupPolicy::IsEnabledOrNotConfigured(TogglePolicy policy) const
+    {
+        return GetState(policy) != PolicyState::Disabled;
+    }
+
+    static std::unique_ptr<GroupPolicy> s_groupPolicy;

```

Comment:
>static std::unique_ptr<GroupPolicy> s_groupPolicy; [](start = 4, length = 50)

It looks like you are doing it this way to enable testing scenarios.  It isn't thread safe this way though.

I would suggest a test hook be implemented this way:
```
GP& InstanceInternal(std::optional<GroupPolicy*> overridePolicy = {})
{
static GroupPolicy s_groupPolicy = make...;
static GroupPolicy* s_override = nullptr;

if (overridePolicy)
{
s_override = oP.value()
}

return (s_override ? *s_override : s_gP);
}

Instance()
{
return InstanceInternal();
}

TEST_HOOK_OVERRIDE_GP(GP* gp)
{
  InstanceInternal(gp);
}
```

Can also make the internal function the basic version when test hooks are disabled.